### PR TITLE
Data utils cleanup

### PR DESCRIPTION
**Pull request by jhamman**

### CODE REVIEW

File: cmip6_downscaling/data/cmip.py [hunk 1]
```diff
@@ -75,17 +106,12 @@ def load_cmip(
                 .to_list()
             )
 
+            # Q: why are we using the era5 storage_options
```
Comment (norlandrhagen) [inline cmip6_downscaling/data/cmip.py:109]: not sure, but maybe @orianac knows?

File: cmip6_downscaling/data/cmip.py [hunk 2]
```diff
@@ -99,95 +125,52 @@ def load_cmip(
...
-    if ds.lat[0] < ds.lat[-1]:
-        ds = ds.reindex({"lat": ds.lat[::-1]})
-    ds = maybe_drop_band_vars(ds)
-    if 'height' in ds:
-        ds = ds.drop('height')
-    ds = ds.squeeze(drop=True)
-    return ds
-
-
 def get_gcm(
```
Comment (norlandrhagen) [inline cmip6_downscaling/data/cmip.py:128]: Should these funcs be passed the run_parameters object?

File: cmip6_downscaling/data/cmip.py [hunk 3]
```diff
@@ -99,95 +125,52 @@ def load_cmip(
...
         End year of train/historical period
-    predict_period_start: str
+    predict_period_start : str
         Start year of predict/future period
-    predict_period_end: str
+    predict_period_end : str
         End year of predict/future period
-    chunking_approach: Optional[str]
-        'full_space', 'full_time', or None
+
```
Comment (norlandrhagen) [inline cmip6_downscaling/data/cmip.py:156]: bbox and cache_within_rechunk are missing docstrings

File: cmip6_downscaling/data/cmip.py [hunk 4]
```diff
@@ -99,95 +125,52 @@ def load_cmip(
...
 def get_gcm(
     gcm: str,
     scenario: str,
     variables: Union[str, List[str]],
     train_period: slice,
     predict_period: slice,
     bbox,
-    chunking_approach: Optional[str] = None,
-    cache_within_rechunk: Optional[bool] = True,
+    cache_within_rechunk: bool = True,
```
Comment (norlandrhagen) [inline cmip6_downscaling/data/cmip.py:135]: I think this arg can be removed.

**Review by norlandrhagen — COMMENTED** [review]
Looks like a good cleanup!

File: cmip6_downscaling/data/cmip.py (hunk 2 above)
Comment (jhamman) [reply cmip6_downscaling/data/cmip.py:128]: I'd like to keep the run_parameters object inside the task/flow scope.

**Pull request merged**